odoo/addons/base/models/ir_http.py

```diff
@@ -201,6 +201,30 @@ def _match(cls, path_info):
     def _get_public_users(cls):
         return [request.env['ir.model.data']._xmlid_to_res_model_res_id('base.public_user')[1]]
 
+    @classmethod
+    def __get_http_authorization_bearer_token(cls):
+        # werkzeug>=2.3
+        authorization = request.httprequest.authorization
+        if authorization and authorization.type == 'bearer':
+            return getattr(authorization, 'token', authorization.get('token'))
+        # werkzeug<2.3, bearer is unsupported, check header directly
+        header = request.httprequest.headers.get("Authorization")
+        if header and (m := re.match(r"^bearer\s+(.+)$", header, re.IGNORECASE)):
+            return m.group(1)
+        # not found
+        if not request.env.uid:
+            raise AccessDenied('User not authenticated, use the "Authorization" header')
+        return None
+
+    @classmethod
+    def _auth_method_bearer(cls):
+        if token := cls.__get_http_authorization_bearer_token():
+            uid = request.env['res.users.apikeys']._check_credentials(scope='rpc', key=token)
+            if not uid:
+                raise AccessDenied()
+            request.update_env(user=uid)
```

Comment:
auth=public can be viewed as "auth=user or auth=public".
I would prefer to consider this discrepancy as an error; because in both session and header we have an discrepancy.